### PR TITLE
[SK-3] Decorate easy components with replaceableComponent

### PR DESCRIPTION
**Pull request by turt2live**
----

**Reviewer**: Take breaks while reviewing this. This is broken into chunks over commits to try and make it easier, though it is probably still too much.

----

### CODE REVIEW

**Comment by turt2live** [comment]
CI will fail until https://github.com/matrix-org/matrix-react-sdk/pull/5732 lands

File: src/components/views/rooms/BasicMessageComposer.tsx [hunk 1]
```diff
@@ -105,6 +106,7 @@ interface IState {
     completionIndex?: number;
 }
 
+@replaceableComponent("views.rooms.BasixMessageEditor")
```
Comment (jryans) [inline src/components/views/rooms/BasicMessageComposer.tsx:109]: Wow, I caught one! 😜 

```suggestion
@replaceableComponent("views.rooms.BasicMessageEditor")
```

File: src/components/views/rooms/ForwardMessage.js [hunk 2]
```diff
@@ -19,8 +19,9 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import { _t } from '../../../languageHandler';
 import {Key} from '../../../Keyboard';
+import {replaceableComponent} from "../../../utils/replaceableComponent";
 
-
+@replaceableComponent("views.rooms.FowardMessage")
```
Comment (jryans) [inline src/components/views/rooms/ForwardMessage.js:24]:
```suggestion
@replaceableComponent("views.rooms.ForwardMessage")
```

File: src/components/views/voip/DialPad.tsx [hunk 3]
```diff
@@ -59,6 +60,7 @@ interface IProps {
     onDialPress?: (string) => void;
 }
 
+@replaceableComponent("views.voip.Dialpad")
```
Comment (jryans) [inline src/components/views/voip/DialPad.tsx:63]:
```suggestion
@replaceableComponent("views.voip.DialPad")
```

File: src/components/views/voip/DialPadModal.tsx [hunk 4]
```diff
@@ -34,6 +35,7 @@ interface IState {
     value: string;
 }
 
+@replaceableComponent("views.voip.DialpadModal")
```
Comment (jryans) [inline src/components/views/voip/DialPadModal.tsx:38]:
```suggestion
@replaceableComponent("views.voip.DialPadModal")
```

**Review by jryans — APPROVED** [review]
Wow, fun! 😜 Looks good overall, I trust you can apply the fixes without another round.

I might recommend merging this after the RC tomorrow, if that's okay with you, just in case ... something strange happens.

File: src/components/views/rooms/MessageComposer.js [hunk 5]
```diff
@@ -32,6 +32,7 @@ import {UIFeature} from "../../../settings/UIFeature";
 import WidgetStore from "../../../stores/WidgetStore";
 import {UPDATE_EVENT} from "../../../stores/AsyncStore";
 import ActiveWidgetStore from "../../../stores/ActiveWidgetStore";
+import {replaceableComponent} from "../../../utils/replaceableComponent";
 
 function ComposerAvatar(props) {
     const MemberStatusMessageAvatar = sdk.getComponent('avatars.MemberStatusMessageAvatar');
```
Comment (t3chguy) [inline src/components/views/rooms/MessageComposer.js:38]: So now that things like `MemberStatusMessageAvatar` are a replacableComponent can we ditch getComponent?

**Comment by turt2live** [comment]
@jryans fyi https://github.com/matrix-org/matrix-react-sdk/pull/5734/commits/375ffafda647a47f9390b272f93b84ca2d8571a0 is new

**Review by jryans — APPROVED** [review]
We should endeavour to eventually understand why this is needed and improve it, but for now it's okay.

**Pull request merged**